src/Symfony/Component/Cache/Traits/FilesystemCommonTrait.php

```diff
@@ -112,7 +112,13 @@ private function write(string $file, string $data, int $expiresAt = null)
                 touch($this->tmp, $expiresAt ?: time() + 31556952); // 1 year in seconds
             }
 
-            return rename($this->tmp, $file);
+            $success = rename($this->tmp, $file);
```

Comment:
what about this instead?
```suggestion
            if (!@rename($this->tmp, $file)) {
                unlink($this->tmp);
                
                return false;
            }
```

Reply:
See https://github.com/php/php-src/blob/master/main/streams/plain_wrapper.c#L1272-L1371

Internally, when using the `rename()` function, PHP will do a `copy()` + `unlink()`, but will not return false if `unlink()` fails. So it is potentially possible to get a success on rename operation, but to still have the temporary file here.

I hesitate while coding this fix. As it is really an edge case, it can be ignored, in order to optimize performances for the most common case.